File: 프로그래머스 풀이/lv1_약수의_개수와_덧셈.java
class Solution {
  public int solution(int left, int right) {
      int answer = 0;
      double sq;
      while ( left <= right ) {
          sq = Math.sqrt(left);
          int val = (int) sq;
          if ( val*val == left )  answer -= left;
          else    answer += left;
          left++;
      }
      return answer;
  }
}

File: 프로그래머스 풀이/lv1_약수의_개수와_덧셈.py
# version 1
import math
def solution(left, right):
    sqrt_left = math.ceil(math.sqrt(left))
    sqrt_right = math.ceil(math.sqrt(right))
    
    if right == sqrt_right ** 2 :
        sqrt_right += 1
        
    m_sum = [i**2 for i in range(sqrt_left, sqrt_right)]
    ssum = 0
    
    for i in range(left, right+1) :
        ssum += i

    return ssum - 2 * sum(m_sum)

# version 2 : 다른 유저가 만든 코드
def solution(left, right):
    return sum([-n if int(n ** 0.5) == n ** 0.5 else n for n in range(left, right+1)])